src/Raven.Server/Documents/Sharding/ShardedRequestHandler.cs

```diff
@@ -79,7 +82,26 @@ public override async Task WaitForIndexToBeApplied(TransactionOperationContext c
                     var db = await task;
                     tasks.Add(db.RachisLogIndexNotifications.WaitForIndexNotification(index, ServerStore.Engine.OperationTimeout));
                 }
-                await tasks.WhenAll();
+                await Task.WhenAll(tasks);
+            }
+        }
+
+        protected async Task WaitForExecutionOfRaftCommands(JsonOperationContext context, List<long> raftIndexIds)
+        {
```

Comment:
add a ToDo that this should be removed after we migrate to ShardedExecutor

Reply:
done